miden/Cargo.toml

```diff
@@ -62,6 +62,7 @@ assert_cmd = "2.0"
 blake3 = "1.3"
 criterion = "0.4"
 escargot = "0.5.7"
+miden-test-utils = { package = "miden-test-utils", path = "../miden-test-utils", version = "0.1" }
 num-bigint = "0.4"
 predicates = "2.1.5"
 processor = { package = "miden-processor", path = "../processor", version = "0.5", features = ["internals"], default-features = false }
```

Comment:
nit: let's name it `miden-test` - i.e., this line could be:
```
miden-test = { package = "miden-test-utils", path = "../miden-test-utils", version = "0.1" }
```

Also, should we get rid of some dependencies here and just re-export them from `miden-test-utils`. I think the following dependencies could probably be moved there:
- `processor`.
- `proptest`
- `rand-utils`.
- `test-case`.
- Maybe `vm-core`?
- Maybe Winterfell crates?

Reply:
Yes we can do that - makes it cleaner. I'll send commit with those changes.

Reply:
As I attempted to do it, I understand we don't need to do this. It's much better to comprehend for someone to understand what dependency is coming from where - easy to understand if `rand_array` comes from `winter-rand-utils` rather than `miden-test-utils`.

Reply:
I actually think it would be good if the user doesn't need to be aware that things like `winter-rand-utils` even exist. The idea of `miden-test-utils` is that it would provide a set of tools for testing Miden VM programs and I think generating random values is a part of that.

I think an argument can be made about importing `proptest` and `testcase` separately - but I still think it is a good idea to include them as well as it simplifies overall testing.